readline/line.go

```diff
@@ -183,26 +183,72 @@ func (rl *Instance) deleteToEnd() {
 	rl.line = rl.line[:rl.pos]
 }
 
-func (rl *Instance) emacsForwardWord(tokeniser tokeniser) (adjust int) {
-	// when emacs has more specific stuff, move this in a file with then
-	split, index, pos := tokeniser(rl.line, rl.pos)
-	if len(split) == 0 {
-		return
+func (rl *Instance) isAlphabetic(r rune) (b bool) {
+	switch {
+	case 97 <= r && r <= 122:
+		return true
+	case 65 <= r && r <= 90:
+		return true
+	default:
+		return false;
 	}
+}
 
-	word := rTrimWhiteSpace(split[index])
+func (rl *Instance) emacsForwardWord() (adjust int) {
+	index := rl.pos
+	end := len(rl.line)
 
-	switch {
-	case len(split) == 0:
-		return
-	case index == len(split)-1 && pos >= len(word)-1:
-		return
-	case pos >= len(word)-1:
-		word = rTrimWhiteSpace(split[index+1])
-		adjust = len(split[index]) - pos
-		adjust += len(word)
-	default:
-		adjust = len(word) - pos
+	if ((index > end) || index == end) {
+		return 0
+	}
+
+	// goto first nonalphabet char if not already on one
+	if rl.isAlphabetic(rl.line[index]) == false {
+		index++
+		for ;index < end;index++ {
```

Comment:
also idk if this is bad but i did it anyway, can make it look nicer i guess by adding index++ to end of loop